doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,574 @@
+If an AS receives a FABRID packet but does not support FABRID, it treats the packet as a normal SCION packet.
+In both cases, all the logic of a normal SCION packet will be applied too.
+The router determines whether the SCION packet is a FABRID packet as follows:
+
+.. image:: fig/FABRID/FABRIDActivation.png
+
+If the router supports FABRID and the SCION packet contains the FABRID HBH extension, the router is going to verify the
+correctness of the current FABRID Hop-validation-field using either the AS-to-AS or AS-to-Host DRKey and verifies whether
+the encrypted policy index matches a valid FABRID policy.
+If this is the case, the router will update the FABRID HVF to HVFVerified, see the :ref:`Header fields computation <fabrid-formulas>`,
+and route the packet over an intra-AS path matching the provided FABRID policy.
+All intra-AS paths are configured by the AS operator, and are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path and policy constraints. A detailed explanation on how endhost applications can find such paths is given in the section :ref:`Exposing policies to the end hosts <endhost_policy_selection>`. Once a path has been found, with specific policies for each hop in the path, the path and an array containing one policy per hop is given to the FABRID snet implementation. The snet implementation then constructs the FABRID packet by automatically requesting the necessary DRKeys and computing the hop validation fields. The packet can then be sent to the border router for further forwarding. A receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
+With this the endhost is able to create FABRID packets and then send them to the border router for further forwarding.
+The FABRID snet implementation will automatically request the necessary DRKeys and compute the hop validation fields,
+the endhost only has to provide the path and the FABRID policies.
+Then the receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
+
+Control plane
+---------------
+
+Control service
+^^^^^^^^^^^^^^^^^
+
+The control service for FABRID is responsible for maintaining the by the AS-operator configured FABRID policies, intra-AS paths,
+and making them accessible for the routers, the endhosts and other remote control services.
+The policies are defined between interface pairs and for the last AS on the path also per interface - IP range pair.
+Through gRPC, border routers can query the control service for the list of supported policies,
+as well as the mapping from policies to MPLS labels.
+Policies are disseminated to remote ASes through PCBs, which clients in the AS can query from their Path Servers.
```

Comment:
Policies are disseminated -> explain in more detail what exactly is disseminated ("policy" is too general)

Reply:
This is under "PCB dissemination", perhaps a reference to there?